code/game/objects/items/stacks/stack_recipe.dm

```diff
@@ -28,8 +28,84 @@
 	src.window_checks = window_checks
 	src.cult_structure = cult_structure
 
-/datum/stack_recipe/proc/post_build(obj/item/stack/S, obj/result)
-	return
+/// Returns TRUE if the recipe can be built, otherwise returns FALSE. This proc is only meant as a series of tests to check if construction is possible; the actual creation of the resulting atom should be handled in do_build()
+/datum/stack_recipe/proc/try_build(mob/user, obj/item/stack/S, multiplier)
+	if(S.get_amount() < req_amount * multiplier)
+		if(req_amount * multiplier > 1)
+			to_chat(user, "<span class='warning'>You haven't got enough [S] to build \the [req_amount * multiplier] [title]\s!</span>")
+		else
+			to_chat(user, "<span class='warning'>You haven't got enough [S] to build \the [title]!</span>")
+		return FALSE
+
+	if(window_checks && !valid_window_location(get_turf(S), user.dir))
+		to_chat(user, "<span class='warning'>\The [title] won't fit here!</span>")
+		return FALSE
+
+	if(one_per_turf && (locate(result_type) in get_turf(S)))
+		to_chat(user, "<span class='warning'>There is another [title] here!</span>")
+		return FALSE
+
+	if(on_floor && !issimulatedturf(get_turf(S)))
+		to_chat(user, "<span class='warning'>\The [title] must be constructed on the floor!</span>")
+		return FALSE
+	if(on_floor_or_lattice && !(issimulatedturf(get_turf(S)) || locate(/obj/structure/lattice) in get_turf(S)))
+		to_chat(user, "<span class='warning'>\The [title] must be constructed on the floor or lattice!</span>")
+		return FALSE
+
+	if(cult_structure)
+		if(user.holy_check())
+			return
+		if(!is_level_reachable(user.z))
+			to_chat(user, "<span class='warning'>The energies of this place interfere with the metal shaping!</span>")
+			return
+		if(locate(/obj/structure/cult) in get_turf(S))
+			to_chat(user, "<span class='warning'>There is a structure here!</span>")
+			return FALSE
+
+	if(time)
+		to_chat(user, "<span class='notice'>Building [title]...</span>")
+		if(!do_after(user, time, target = S.loc))
+			return FALSE
+
+	if(cult_structure && locate(/obj/structure/cult) in get_turf(S)) //Check again after do_after to prevent queuing construction exploit.
+		to_chat(user, "<span class='warning'>There is a structure here!</span>")
+		return FALSE
+
+	if(S.get_amount() < req_amount * multiplier)
+		return
```

Comment:
No need to return FALSE here and in the other places inside this proc?

Reply:
No, but it probably should.  Fixed.